`Update CODEOWNERS` workflow is failing

Is something wrong with CI or publishing infrastructure? Yes.

The workflow is failing with `Error: Cannot find module 'types-publisher'`.

PR #45101 removed that package from the package.json.  @sandersn Should it be added back?

Additionally, why is this workflow running on everyone's repo? @orta Can that be managed?  I only know it is running because of the failures.

--- a/.github/workflows/UpdateCodeowners.yml
+++ b/.github/workflows/UpdateCodeowners.yml
@@ -12,6 +12,7 @@ on:
 jobs:
   update:
     runs-on: ubuntu-latest
+    if: github.repository == 'DefinitelyTyped/DefinitelyTyped'
 
     steps:
       - uses: actions/checkout@v2

--- a/package.json
+++ b/package.json
@@ -36,5 +36,5 @@
             "_comment": "This will remove husky from when we started migrating to use prettier",
             "pre-commit": "npm uninstall husky"
         }
-    },
+    }
 }

--- a/package.json
+++ b/package.json
@@ -24,6 +24,8 @@
     },
     "devDependencies": {
         "@definitelytyped/dtslint-runner": "latest",
+        "@definitelytyped/definitions-parser": "latest",
+        "@definitelytyped/utils": "latest",
         "danger": "^10.1.1",
         "dtslint": "latest",
         "prettier": "^2.0.2",
@@ -35,8 +37,4 @@
             "pre-commit": "npm uninstall husky"
         }
     },
-    "dependencies": {
-        "@definitelytyped/definitions-parser": "0.0.40",
-        "@definitelytyped/utils": "0.0.40"
-    }
 }

--- a/package.json
+++ b/package.json
@@ -34,5 +34,9 @@
             "_comment": "This will remove husky from when we started migrating to use prettier",
             "pre-commit": "npm uninstall husky"
         }
+    },
+    "dependencies": {
+        "@definitelytyped/definitions-parser": "0.0.40",
+        "@definitelytyped/utils": "0.0.40"
     }
 }

--- a/scripts/update-codeowners.js
+++ b/scripts/update-codeowners.js
@@ -1,9 +1,10 @@
 /// <reference lib="esnext.asynciterable" />
 // Must reference esnext.asynciterable lib, since octokit uses AsyncIterable internally
 const cp = require("child_process");
+const os = require("os");
 const Octokit = require("@octokit/rest");
-const { AllPackages, getDefinitelyTyped, loggerWithErrors,
-        parseDefinitions, parseNProcesses, clean } = require("types-publisher");
+const { AllPackages, getDefinitelyTyped, parseDefinitions, clean } = require("@definitelytyped/definitions-parser");
+const { loggerWithErrors } = require("@definitelytyped/utils");
 const { writeFile } = require("fs-extra");
 
 async function main() {
@@ -12,7 +13,7 @@ async function main() {
 
     clean();
     const dt = await getDefinitelyTyped(options, log);
-    await parseDefinitions(dt, { nProcesses: parseNProcesses(), definitelyTypedPath: "." }, log);
+    await parseDefinitions(dt, { nProcesses: os.cpus.length, definitelyTypedPath: "." }, log);
     const allPackages = await AllPackages.read(dt);
     const typings = allPackages.allTypings();
     const maxPathLen = Math.max(...typings.map(t => t.subDirectoryPath.length));

